### DISCUSSION TITLE
Moving Faders Without Touch

### DISCUSSION THREAD

**Discussion by markwinger** [tttapa/Control-Surface-Motor-Fader#22] [Ideas]
I have done some similar projects but I am looking at using this project as a base for a new one.  I used teensy's before, but this mega328 looks like a nice approach.  I have a bunch of faders that do not have touch tracks and want to use them.  The project is for a live console, not a DAW.  Automation is not the goal, snapshot recall and multiple views is.

So I would like to remove the touch ability, and have midi controllers sent the the host when a fader is moved, and move the faders to the target location when the host is recalling a snapshot or different view.

So the issue is to trigger midi events when a fader is moved instead of when it is touched.  Also the motors must turn off quickly after positioning.

I have been looking at this code with this in mind and wondering if anyone sees any difficulty in this approach and any advice before start actually trying to modify the code to work this way.

Thanks.

**Comment by tttapa** [comment]
Stripping out the touch-related stuff from `main.cpp` should be straightforward enough. Then you'll have to change the logic to disable the controller after the setpoint has been reached:

https://github.com/tttapa/Control-Surface-Motor-Fader/blob/fe8a54f912ea0bac2d75647e6f40e23807e00823/Motor-Controller/Controller.hpp#L77-L89

As you can see, the controller is turned off if the error is low enough, and enough time has passed since the last setpoint change. (`activityCount` counts the number of time steps since the last setpoint change.)  
If the motor is turned off, and the error becomes too large (larger than `errThres` in absolute value), the motor is turned on again (with some hysteresis to avoid oscillations). I guess in the use case you describe, you'd have to remove this second case where the motor is turned on again, i.e. only a setpoint change should turn on the motor.

**Comment by markwinger** [comment]
Thanks, that is exactly the approach I am working on.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
